web/client/components/map/leaflet/MeasurementSupport.jsx

```diff
@@ -2,21 +2,217 @@ const PropTypes = require('prop-types');
 const React = require('react');
 const assign = require('object-assign');
 var L = require('leaflet');
-const {slice} = require('lodash');
-const {reproject, calculateAzimuth, calculateDistance, transformLineToArcs} = require('../../../utils/CoordinatesUtils');
+const {
+    slice
+} = require('lodash');
+const {
+    reproject,
+    calculateAzimuth,
+    calculateDistance,
+    transformLineToArcs
+} = require('../../../utils/CoordinatesUtils');
+const {
+    sqmTosqft,
+    sqmTosqnm,
+    sqmTosqmi,
+    sqmTosqkm,
+    getFormattedBearingValue
+} = require('../../../utils/MeasureUtils');
 require('leaflet-draw');
 
+let defaultPrecision = {
+    km: 2,
+    ha: 2,
+    m: 2,
+    mi: 2,
+    ac: 2,
+    yd: 0,
+    ft: 0,
+    nm: 2
+};
+
+
+/** @method readableDistance(distance, units): string
+ * Converts a metric distance to one of [ feet, nauticalMile, metric or yards ] string
+ *
+ * @alternative
+ * @method readableDistance(distance, isMetric, useFeet, isNauticalMile, precision, options): string
+ * Converts metric distance to distance string.
+ * The value will be rounded as defined by the precision option object.
+ * this override is necesary due to the customization on how the measure label is presented and for adding bearing support
+*/
+L.GeometryUtil.readableDistance = (distance, isMetric, isFeet, isNauticalMile, precision, options) => {
+    let distanceStr;
+    let units;
+    let d = distance;
+    let p = L.Util.extend({}, defaultPrecision, precision);
+
+    if (isMetric) {
+        units = typeof isMetric === 'string' ? isMetric : 'metric';
+    } else if (isFeet) {
+        units = 'feet';
+    } else if (isNauticalMile) {
+        units = 'nauticalMile';
+    } else {
+        units = 'yards';
+    }
+    if (options.geomType === 'Bearing') {
+        return options.bearing;
+    }
+    switch (units) {
+        // show meters when distance is < 1km, then show km
+        case 'metric':
+            if (options.distinctMeasure) {
```

Comment:
Why distinctMeasure? Explain the meaning please

Reply:
i meant when you use a treshold to switch from meters and kilometers for example (like leaflet).
i'll change it to useTreshold

Reply:
ok, that's more clear